bitsail-components/bitsail-component-metrics/bitsail-component-metrics-prometheus/src/main/java/com/bytedance/bitsail/component/metrics/prometheus/impl/PrometheusMetricReporter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 Bytedance Ltd. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.bytedance.bitsail.component.metrics.prometheus.impl;
+
+import com.bytedance.bitsail.common.BitSailException;
+import com.bytedance.bitsail.common.configuration.BitSailConfiguration;
+import com.bytedance.bitsail.component.metrics.prometheus.AbstractPrometheusReporter;
+import com.bytedance.bitsail.component.metrics.prometheus.error.PrometheusErrorCode;
+import com.bytedance.bitsail.component.metrics.prometheus.option.PrometheusOptions;
+
+import io.prometheus.client.dropwizard.DropwizardExports;
+import io.prometheus.client.exporter.HTTPServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+public class PrometheusMetricReporter extends AbstractPrometheusReporter {
```

Comment:
what does this do? too many inheritance may bring in some unnecessary complexity. abstract the class when necessary in the future is also ok

Reply:
prometheus support pull&push mode, push mode use prometheus gateway reporter, pull mode use prometheus reporter, when use pull mode, user should make sure prometheus server and BitSail work network segment, push mode don't need it.